src/viem/prepareTransactions.ts

```diff
@@ -94,8 +104,8 @@ export async function tryEstimateTransaction({
 
   // TODO maybe cache this? and add static padding when using non-native fee currency
   try {
-    tx.gas = await publicClient.celo.estimateGas({
-      ...tx,
+    tx.gas = await estimateGas(client, {
+      ...(tx as TransactionRequestBase),
```

Comment:
If you give `tx` the type `TransactionRequestBase` explicitly, there is a TS error. It would be great to avoid type casting to a type that we know doesn't actually work for the data.

Reply:
I tinkered with this and found something that kinda works:
- use `TransactionRequest` type for `tx` (e.g. `const tx: TransactionRequest = { ... `)
- estimateGas doesn't accept 'cip64' as a "type" on tx. Instead of covering this up with a type cast, explicitly omit type if it's 'cip64'. `type: tx.type === 'cip64' ? undefined : tx.type,`. Should probably report this to viem and link to the issue in a code comment as well.
- update the Logger.info with this: `feeCurrency: 'feeCurrency' in tx ? tx.feeCurrency : undefined,` to avoid a type error

Reply:
I think this makes our type workarounds more transparent and readable, while keeping the static typing guarantees for all other fields in `tx` besides the troublesome `type` field

Reply:
well anyway. looks like this one's actually pretty hard. I'd like to minimize type casting, but we may have to settle for some hacky workaround for the sake of expediency

Reply:
Yes exactly. I try to avoid casting as much as possible, until it's too "much".
Thanks for trying.